src/main/java/com/minecolonies/coremod/permissions/ColonyPermissionEventHandler.java

```diff
@@ -206,6 +205,57 @@ public void on(final PlayerInteractEvent event)
     }
 
     /**
+     * PlayerInteractEvent.EntityInteract handler.
+     * <p>
+     * Check, if a player right clicked an entity.
+     * Deny if:
+     * - If the entity is in colony
+     * - player has not permission
+     *
+     * @param event PlayerInteractEvent
+     */
+    @SubscribeEvent
+    public void on(final PlayerInteractEvent.EntityInteract event)
+    {
+        final EntityPlayer playerIn = event.getEntityPlayer();
+        if (Configurations.enableColonyProtection && colony.isCoordInColony(playerIn.getEntityWorld(), playerIn.getPosition()))
```

Comment:
Also don't this it's a big deal, but you check if the player is standing in the colony, not if the citizen is (or if it even is a citizen, etc)